cookbooks/omnibus-supermarket/recipes/database.rb

```diff
@@ -19,21 +19,31 @@
 
 include_recipe 'omnibus-supermarket::config'
 
-# The enterprise_pg resources use the CLI to create databases and users. Set
-# these environment variables so the commands have the correct connection
-# settings.
+create_superuser = node['supermarket']['database']['user'] != node['supermarket']['postgresql']['username']
 
-ENV['PGHOST'] = node['supermarket']['database']['host']
-ENV['PGPORT'] = node['supermarket']['database']['port'].to_s
+pg_server_env = {
+  'PGHOST' => node['supermarket']['database']['host'],
+  'PGPORT' => node['supermarket']['database']['port'].to_s
+}
+
+supermarket_env = {
+  'PGUSER' => node['supermarket']['database']['user'],
+  'PGPASSWORD' => node['supermarket']['database']['password'],
+  'RAILS_ENV' => 'production'
+}.merge(pg_server_env)
+
+# enterprise_pg* resources will use the postgresql user, not the database user.
+# If we supply the host/port ENV variables in combined mode it will attempt
+# to use a TCP socket instead of a Unix socket, which we don't want to do
+# because the Chef Server postgresql user is local only.
+
+pg_server_env.each_pair { |k, v| ENV[k] = v } unless combined_mode?
+node['supermarket']['database']['password']
 
 enterprise_pg_user node['supermarket']['database']['user'] do
   superuser true
   password node['supermarket']['database']['password'] || ''
-  # If the database user is the same as the main postgres user, don't create it.
-  not_if do
-    node['supermarket']['database']['user'] ==
-      node['supermarket']['postgresql']['username']
-  end
+  only_if { create_superuser }
```

Comment:
I don't really see the point of extracting this out into a variable if it's only going to be used in one place.